app/api/v1/endpoints/telegram.py

```
@@ -564,15 +564,27 @@ async def _verify_telegram_webhook(request: Request) -> bool:
         signature = request.headers.get('X-Telegram-Bot-Api-Secret-Token')
-        return signature == webhook_secret
+        is_valid = signature == webhook_secret
+        
+        if not is_valid:
+            logger.warning("Webhook verification failed", 
+                         expected_secret=webhook_secret[:8] + "...", 
+                         received_signature=signature[:8] + "..." if signature else "None")
+        else:
+            logger.info("Webhook verification passed")
+            
+        return is_valid
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)